runner/runners/README.md

```diff
@@ -0,0 +1,31 @@
+# Runner Impl Design #
```

Comment:
It would be great if the interface/struct hierarchy could be outlined visually in one block of text near the top.